File: Providers/ResultLayoutProvider.vb
'************************************************************************************
' STSimStockFlow: A .NET library for simulating stocks and flows
'
' Copyright © 2009-2015 ApexRMS.
'
'************************************************************************************

Imports SyncroSim.Core
Imports SyncroSim.Core.Forms
Imports System.Reflection

<ObfuscationAttribute(Exclude:=True, ApplyToMembers:=False)>
Class ResultLayoutProvider
    Inherits LayoutProvider

    Protected Overrides Sub ModifyLayout(layout As SyncroSimLayout)

        Dim ExportGroup As SyncroSimLayoutItem = layout.Items.GetItem("Export")

        If (ExportGroup IsNot Nothing) Then

            Dim ReportsGroup As SyncroSimLayoutItem = ExportGroup.Items.GetItem("Reports")
            Dim MapsGroup As SyncroSimLayoutItem = ExportGroup.Items.GetItem("Maps")

            If (ReportsGroup IsNot Nothing) Then

                ReportsGroup.Items.Add(New SyncroSimLayoutItem("stsim-stockflow:summary-stock-report", "Stocks", False))
                ReportsGroup.Items.Add(New SyncroSimLayoutItem("stsim-stockflow:summary-flow-report", "Flows", False))

            End If

            If (MapsGroup IsNot Nothing) Then

                MapsGroup.Items.Add(New SyncroSimLayoutItem("stsim-stockflow:stock-raster-map", "Stock Types", False))
                MapsGroup.Items.Add(New SyncroSimLayoutItem("stsim-stockflow:flow-raster-map", "Flow Types", False))
                MapsGroup.Items.Add(New SyncroSimLayoutItem("stsim-stockflow:stock-group-raster-map", "Stock Groups", False))
                MapsGroup.Items.Add(New SyncroSimLayoutItem("stsim-stockflow:flow-group-raster-map", "Flow Groups", False))

            End If

        End If

    End Sub

End Class

File: Providers/ResultLayoutProvider.cs
// STSimStockFlow: A SyncroSim Package for the ST-Sim Stocks and Flows Add-In.
// Copyright © 2007-2018 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.

using SyncroSim.Core;
using SyncroSim.Core.Forms;
using System.Reflection;

namespace SyncroSim.STSimStockFlow
{
	[ObfuscationAttribute(Exclude=true, ApplyToMembers=false)]
	internal class ResultLayoutProvider : LayoutProvider
	{
		protected override void ModifyLayout(SyncroSimLayout layout)
		{
			SyncroSimLayoutItem ExportGroup = layout.Items.GetItem("Export");

			if (ExportGroup != null)
			{
				SyncroSimLayoutItem ReportsGroup = ExportGroup.Items.GetItem("Reports");
				SyncroSimLayoutItem MapsGroup = ExportGroup.Items.GetItem("Maps");

				if (ReportsGroup != null)
				{
					ReportsGroup.Items.Add(new SyncroSimLayoutItem("stsim-stockflow:summary-stock-report", "Stocks", false));
					ReportsGroup.Items.Add(new SyncroSimLayoutItem("stsim-stockflow:summary-flow-report", "Flows", false));
				}

				if (MapsGroup != null)
				{
					MapsGroup.Items.Add(new SyncroSimLayoutItem("stsim-stockflow:stock-raster-map", "Stock Types", false));
					MapsGroup.Items.Add(new SyncroSimLayoutItem("stsim-stockflow:flow-raster-map", "Flow Types", false));
					MapsGroup.Items.Add(new SyncroSimLayoutItem("stsim-stockflow:stock-group-raster-map", "Stock Groups", false));
					MapsGroup.Items.Add(new SyncroSimLayoutItem("stsim-stockflow:flow-group-raster-map", "Flow Groups", false));
				}
			}        
		}
	}
}